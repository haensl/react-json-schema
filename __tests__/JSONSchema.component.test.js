import { render } from '@testing-library/react';
import JSONSchema from '../';

describe('JSONSchema component', () => {
  it('renders without crashing', () => {
    expect(render.bind(null, <JSONSchema />))
      .not
      .toThrow();
  });

  describe('no JSON', () => {
    it('renders as expected', () => {
      const { container } = render(<JSONSchema />);
      expect(container.firstChild)
        .toMatchSnapshot();
    });
  });

  describe('given JSON', () => {
    it('renders as expected', () => {
      const org = {
        '@context': 'https://schema.org/',
        '@type': 'Organization',
        '@id': 'https://acme.com',
        url: 'https://acme.com',
        name: 'ACME'
      };

      const { container } = render(<JSONSchema json={ org } />);

      expect(container.firstChild)
        .toMatchSnapshot();
    });
  });
});

