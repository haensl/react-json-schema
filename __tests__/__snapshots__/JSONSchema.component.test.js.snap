// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSONSchema component given JSON renders as expected 1`] = `
<script
  type="application/ld+json"
>
  {"@context":"https://schema.org/","@type":"Organization","@id":"https://acme.com","url":"https://acme.com","name":"ACME"}
</script>
`;

exports[`JSONSchema component no JSON renders as expected 1`] = `null`;
